### ISSUE TITLE
Add a release notes modal

### ISSUE THREAD

**Issue by PetrCala** [PetrCala/Kiroku#92]

**Comment by PetrCala** [comment]
## Rough solution outline — in-app "What's New" modal

> **Scope:** design sketch only, **no implementation / no code changes**. This maps the general direction against the infra we already have, with real file paths to anchor it. Not a quick win.

First, to clear up a naming collision: the existing [`release-notes` skill](.claude/skills/release-notes) generates **store-facing** "What's New" copy (App Store / Play listing text) by diffing the last publicly shipped store version against master. That's a build-time authoring tool for the store listing — it does **not** render anything in the app. This issue is the separate, **in-app** modal shown to a user *after* they update. The two are complementary: the skill's per-version prose becomes the *content source* for this modal (see §2).

---

### 1. Trigger logic — "updated to a version whose notes the user hasn't seen"

We already read the app version and gate on versions:

- Current version: `pkg.version` (`package.json`, currently `0.3.16-27`), surfaced as `version` in [src/libs/Validation.ts](src/libs/Validation.ts).
- That file already does semver gating in `validateAppVersion()` using the `semver` lib, and crucially has `cleanSemver()` which strips build metadata: `0.3.16-27` → `0.3.16`. `min_supported_version` / `latest_version` (force-update + update-available) all compare at this **MAJOR.MINOR.PATCH (MMP)** level.

**Key the modal off the MMP version, not the build.** Builds (`-27`, `-28`, …) are not user-facing releases; keying off the full string would re-pop the modal on every internal build bump. So:

```
currentMMP   = cleanSemver(pkg.version)                  // "0.3.16"
lastSeenMMP  = nvp_lastSeenReleaseNotesVersion           // "0.3.15" | undefined
hasNotesFor  = (v) => !!RELEASE_NOTES[v]                 // do we ship content for it?

shouldShow = hasNotesFor(currentMMP)
          && (lastSeenMMP === undefined || semver.gt(currentMMP, lastSeenMMP))
```

Store the "last seen" as a **per-user NVP**, mirroring the existing scalars in [src/ONYXKEYS.ts](src/ONYXKEYS.ts) (`NVP_TERMS_ACCEPTED_VERSION: number`, `NVP_ONBOARDING`, `NVP_LAST_VIEWED_CALENDAR_DATE: Record<...>`):

```
NVP_LAST_SEEN_RELEASE_NOTES_VERSION: 'nvp_lastSeenReleaseNotesVersion'   // value: Semver string
```

Read via `useOnyx(ONYXKEYS.NVP_LAST_SEEN_RELEASE_NOTES_VERSION)`; write optimistically then persist, exactly like `completeOnboarding`/`acceptTerms` do (`Onyx.merge` + `API.write` with `optimisticData` — see [src/libs/actions/Onboarding/index.ts](src/libs/actions/Onboarding/index.ts)). Needs one small kiroku-api endpoint mirroring accept-terms.

**Fresh-install suppression (important):** a brand-new user who onboards *on* `0.3.16` did not "update to" it — they shouldn't see 0.3.16's notes. Cleanest fix: when `completeOnboarding()` runs, stamp `nvp_lastSeenReleaseNotesVersion = currentMMP`. New users start "caught up" and first see notes only after their *next* update. No extra flag needed.

---

### 2. Content source — bundled static, localized via `en.ts` (recommended)

Two options:

**A. Bundled static (recommended).** A typed map keyed by MMP version, e.g. `src/CONST/releaseNotes.ts`, with the actual strings living as i18n keys in [src/languages/en.ts](src/languages/en.ts) (`releaseNotes['0.3.16'].title`, `.items[]`) and mirrored into every locale via the `translate` skill.

**B. Server/config-driven.** Extend the remote `config` object ([src/types/onyx/Config.ts](src/types/onyx/Config.ts), hydrated by OpenApp + the `config`/`configUpdate` Pusher broadcast, read via `useConfig()` in [src/context/global/ConfigContext.tsx](src/context/global/ConfigContext.tsx)) with a `release_notes` blob.

| | Bundled (A) | Server-driven (B) |
|---|---|---|
| Update without an app release | ❌ | ✅ |
| Localization | ✅ via `en.ts` + `translate` skill + the `TranslationContext` test gate | ❌ would have to ship locale-keyed strings server-side, bypassing the whole translation pipeline |
| Offline / first-boot safe | ✅ | ⚠️ depends on config having hydrated |
| Compile-time typed + PR-reviewable | ✅ | ❌ |
| Backend work | none | new field + kiroku-api work |

**Recommend A.** The decisive point: release notes describe the changes *in the build the user just installed* — content and binary ship together, so "staleness" is a non-issue (you never need to edit 0.3.16's notes after 0.3.16 ships). And localization is exactly what `en.ts` + the translate skill + the always-on `TranslationContext` unit test are built for; a config blob would route around all of it. The existing `release-notes` skill already drafts the per-version prose — at release time, paste that into the bundled map and run `translate`.

Keep one cheap **server escape hatch** for later (Phase 2): a `config.release_notes_kill_switch` boolean or `min_release_notes_version`. That's just a flag/version (no localization), so it can live in `config` without the downsides above — lets you suppress the modal remotely if a release's notes are wrong.

---

### 3. UI — reuse the existing modal infra; mount as a guard

Don't invent a modal. Mirror [src/libs/Navigation/guards/TermsReConsentGuard.tsx](src/libs/Navigation/guards/TermsReConsentGuard.tsx) — it is the exact template: a small guard component that reads Onyx, computes `shouldShow`, and returns either `null` or a `<Modal>`.

- New file `src/libs/Navigation/guards/WhatsNewGuard.tsx`.
- Use `@components/Modal` (wraps [src/components/Modal/BaseModal.tsx](src/components/Modal/BaseModal.tsx)). Modal types are in [src/CONST.ts](src/CONST.ts) `MODAL.MODAL_TYPE`. Match the house style with `BOTTOM_DOCKED`, or go responsive like `ConfirmModal` (`shouldUseNarrowLayout ? BOTTOM_DOCKED : CENTERED`). Unlike terms, this one **is** dismissable — wire real `onClose`/`onDismiss`.
- Content: a small `ReleaseNotesContent` (version header + bulleted items from i18n + a single "Got it" button) using existing `Button`/`Text`/`ScrollView` primitives.
- **Mount point:** in [src/libs/Navigation/AppNavigator/AuthScreens.tsx](src/libs/Navigation/AppNavigator/AuthScreens.tsx) right after `<TermsReConsentGuard />` (~line 356), alongside `<OnboardingGuard />`. Mounting inside `AuthScreensContent` guarantees: only authenticated users, splash already hidden, and it's *outside* `RootStack.Navigator` so dismissal never touches the nav stack.

---

### 4. Dismissal, "don't show again", and not stacking on the first-run gates

**Dismiss = mark seen.** On close, stamp `nvp_lastSeenReleaseNotesVersion = currentMMP` (optimistic `Onyx.merge` + `API.write`). One acknowledgement per version; it never re-shows for that version.

**Strict ordering so nothing stacks.** All three first-run surfaces (onboarding, terms re-consent, what's-new) are root-level modals, so they must be *serialized*, not rendered simultaneously. `WhatsNewGuard`'s `shouldShow` should reuse the **same readiness gate** as `TermsReConsentGuard` plus a downstream condition:

```
if (!auth?.currentUser) return false;
if (isLoadingApp !== false || userData === undefined) return false;  // wait for OpenApp to settle
if (!hasCompletedOnboarding(userData)) return false;                 // excludes onboarding
if (!hasAcceptedCurrentTerms(userData, config)) return false;        // ← downstream of terms re-consent
return shouldShow(currentMMP, lastSeenMMP);
```

That last line makes What's New strictly *after* terms: terms shows first; only once accepted does What's New evaluate true. Onboarding is already excluded by `hasCompletedOnboarding`. Fresh installs are handled by the onboarding-complete stamp from §1. (A more robust long-term option is a shared "first-run modal queue," but it's overkill for MVP — the two boolean gates serialize cleanly.)

---

### 5. Open questions for the founder + phased plan

**Decisions needed:**
1. **Version granularity** — confirm MMP (`0.3.16`), not build. (Recommend MMP.)
2. **Which versions get a modal** — every release, or only "notable" ones? Recommend *opt-in per version*: a version triggers the modal only if it has a curated entry in the map, so you control frequency and skip minor patch releases.
3. **Fresh-install suppression** — confirm new users should *not* see notes for the version they installed on (recommend yes; stamp on onboarding complete).
4. **Backfill for existing users** — when the feature first ships, every existing user has no NVP, so they'd all see the current version's notes once. Desired ("here's what's new!"), or suppress via a one-time migration stamp? Founder's call.
5. **Content source** — confirm bundled + optional remote kill-switch (§2).
6. **Always dismissable** — recommend yes; it's informational, not legal (contrast terms, which is non-dismissable).

**Phased plan:**
- **MVP (Phase 0):** `NVP_LAST_SEEN_RELEASE_NOTES_VERSION` + `setLastSeenReleaseNotes` action (optimistic + `API.write`) + onboarding-complete stamp; bundled `releaseNotes.ts` for **one** version; `WhatsNewGuard` mirroring `TermsReConsentGuard`, mounted after it; `en.ts` keys only. One kiroku-api endpoint (clone accept-terms). English-only is fine to start.
- **Phase 1:** localize via the `translate` skill across all locales (covered by the `TranslationContext` test); document the `release-notes` skill → bundled-map handoff so each release authoring step fills the entry.
- **Phase 2 (optional):** remote kill-switch / `min_release_notes_version` in `config`; richer content (image, "see full changelog" screen); dismiss analytics.

---

*Generated as a design sketch — intentionally **not** implemented. Anchored on: `Validation.ts` (version gating), `TermsReConsentGuard.tsx` (the guard/modal template), `Config.ts` + `ConfigContext` (remote config), `ONYXKEYS.ts` NVP scalars, and `AuthScreens.tsx` (mount point).*

**Comment by PetrCala** [comment]
## Decisions locked (recommended defaults)

Resolving the 6 open questions from the design sketch above so this is shovel-ready. These are recommended defaults — override any line and I'll adjust the spec.

| # | Decision | Locked choice | Why |
|---|---|---|---|
| 1 | **Version granularity** | **MMP** (`0.3.16`), via `cleanSemver(pkg.version)` | Matches all existing gating in `Validation.ts`; builds (`-27`) aren't user-facing releases, so the modal must not re-pop on build bumps. |
| 2 | **Which versions trigger it** | **Opt-in per version** — a version shows a modal only if it has a curated entry in `releaseNotes.ts` | Founder controls frequency; skip minor patch releases by simply not adding an entry. |
| 3 | **Fresh-install suppression** | **Suppress** — stamp `nvp_lastSeenReleaseNotesVersion = currentMMP` inside `completeOnboarding()` | A new user who onboards *on* a version didn't "update to" it. New users start caught-up; first see notes after their next update. |
| 4 | **Backfill for existing users** | **No migration needed** — show **only the current version's** notes, never a backlog | First build carrying the feature (e.g. `0.3.17`) → existing users updating into it correctly see `0.3.17`'s notes once (`lastSeen === undefined`). Skipping versions (0.3.16 → 0.3.19) only ever shows 0.3.19. |
| 5 | **Content source** | **Bundled static**, localized via `en.ts` + `translate` skill. Remote kill-switch **deferred to Phase 2** | Content ships with the build it describes (no staleness), and localization stays inside the `en.ts` → translate → `TranslationContext`-test pipeline. |
| 6 | **Dismissability** | **Always dismissable** | Informational, not legal — contrast `TermsReConsentGuard` (non-dismissable). Dismiss = mark seen. |

### Resulting trigger shape (final)

Because we show **only the current version** (decision #4), the comparison simplifies — no accumulation, no backlog:

```
currentMMP = cleanSemver(pkg.version)              // "0.3.17"
lastSeen   = nvp_lastSeenReleaseNotesVersion       // "0.3.16" | undefined

shouldShow = !!RELEASE_NOTES[currentMMP]           // we ship notes for this version
          && (lastSeen === undefined || semver.gt(currentMMP, lastSeen))
          //  semver.gt (not !==) so a downgrade never re-shows old notes
```

Combined with the gate ordering from §4 (downstream of onboarding + terms re-consent), and stamp-on-dismiss + stamp-on-onboarding-complete, this is the complete behavioural spec.

### Phase 0 (MVP) build list — for whenever this gets scheduled

1. `ONYXKEYS.NVP_LAST_SEEN_RELEASE_NOTES_VERSION` + its type in `ONYXKEYS.ts`.
2. `setLastSeenReleaseNotes(version)` action — optimistic `Onyx.merge` + `API.write` (clone the `acceptTerms` shape) — and the stamp inside `completeOnboarding()`.
3. One kiroku-api endpoint to persist the NVP (mirror accept-terms).
4. `src/CONST/releaseNotes.ts` map + `releaseNotes.*` keys in `en.ts` (English-only to start; `translate` skill is Phase 1).
5. `src/libs/Navigation/guards/WhatsNewGuard.tsx` mirroring `TermsReConsentGuard.tsx`, mounted right after `<TermsReConsentGuard />` in `AuthScreens.tsx`.

Still **design only** — nothing implemented.

**Comment by PetrCala** [comment]
## Timing: defer implementation until after public launch

Hold this until the app is **publicly live** — don't build it during the TestFlight / pre-launch phase.

**Why:** pre-launch testers already get a "what's new" surface for free. On iOS, TestFlight shows its built-in **"What to Test"** modal on each new build, so an in-app release-notes modal would be redundant (and would double up with it) for exactly the audience we have right now. The in-app modal only earns its keep once we're shipping to real App Store / Play users who have no such system-provided notes.

**Gate:** implement Phase 0 (per the locked spec above) **only after the first public store release**. No action until then — design + decisions are captured here so it can be picked up cold.
